core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -651,7 +652,11 @@ private ExprValue exprConvertTZ(ExprValue startingDateTime, ExprValue fromTz, Ex
    */
   private ExprValue exprDate(ExprValue exprValue) {
     if (exprValue instanceof ExprStringValue) {
-      return new ExprDateValue(exprValue.stringValue());
+      try {
+        return new ExprDateValue(exprValue.stringValue());
+      } catch (SemanticCheckException e) {
+        return ExprNullValue.of();
+      }
```

Comment:
I'm wondering the reason we prefer NULL over semantic exception. If this is true always, probably we should change `ExprDateValue` and `ExprTimeValue` constructor instead?

Reply:
`Null` is to match the return style that MySQL has, returning `Null` for invalid dates. In this PR that is for dates that are invalid like `2022-02-30` as well as dates not matching the format so something like `02-02-2022`

```mysql> select DATE('02-02-2022');
+--------------------+
| DATE('02-02-2022') |
+--------------------+
| NULL               |
+--------------------+
1 row in set, 1 warning (0.00 sec)
```

```
opensearchsql> select DATE('02-02-2022');                                                                                                                                          
fetched rows / total rows = 1/1
+----------------------+
| DATE('02-02-2022')   |
|----------------------|
| null                 |
+----------------------+
```

However, the `Timestamp` function does not return `Null` instead returning the exception. I think a future PR should be made to move `Timestamp` and possibly other functions in line with returning `Null` to follow MySQL implementations.

Reply:
I'm thinking what's benefit of returning NULL or we just want to follow MySQL for compatibility. Do you happen to know in which case this may be helpful? Like there is an illegal date time value in a result set and we don't want to fail the entire request?

Reply:
Just did some test. MySQL did return NULL for any invalid input. PostgreSQL and SQLServer throws exception.